### PR TITLE
Apply ica new

### PR DESCRIPTION
**Pull request by SophieHerbst**
Zut. Again forgot to properly update my master. 
new stuff starts with e54d48d.

### CODE REVIEW

File: 05a-run_ica.py [hunk 1]
```diff
@@ -65,34 +65,34 @@ def run_ica(subject, tsss=config.mf_st_duration):
...
     for ch_type in ['meg', 'eeg']:
         print('Running ICA for ' + ch_type)
 
         if ch_type == 'meg':
-
-            # XXX this does not work on epochs:
-            # meg_maxfilter_rank = epochs_for_ica.copy().pick_types(meg=True).estimate_rank()
-            n_components = 0.999
+            # XXX requires MNE 0.18
+            n_components = (mne.compute_rank(epochs_for_ica.copy()
```
Comment (SophieHerbst) [inline 05a-run_ica.py]: mne.compute_rank forces people to use version 0.18

File: 05a-run_ica.py [hunk 2]
```diff
@@ -65,34 +65,34 @@ def run_ica(subject, tsss=config.mf_st_duration):
...
 
         ica = ICA(method='fastica', random_state=config.random_state,
                   n_components=n_components)
 
         picks = all_picks[ch_type]
-
+        
+        # XXX this breaks:
+        # *** TypeError: %i format: a number is required, not dict
         ica.fit(epochs_for_ica, picks=picks, decim=decim)
```
Comment (SophieHerbst) [inline 05a-run_ica.py]: @massich @agramfort 
I run into an error here that I wasn't able to fix. My hunch is that it came with using mne 0.18. 
Can you help?

  File "<ipython-input-11-906c8a461b22>", line 1, in <module>
    runfile('/Users/sophie/repos/mne-study-template/05a-run_ica.py', wdir='/Users/sophie/repos/mne-study-template')

  File "/Users/sophie/anaconda/envs/mne/lib/python3.6/site-packages/spyder_kernels/customize/spydercustomize.py", line 704, in runfile
    execfile(filename, namespace)

  File "/Users/sophie/anaconda/envs/mne/lib/python3.6/site-packages/spyder_kernels/customize/spydercustomize.py", line 108, in execfile
    exec(compile(f.read(), filename, 'exec'), namespace)

  File "/Users/sophie/repos/mne-study-template/05a-run_ica.py", line 128, in <module>
    parallel(run_func(subject) for subject in config.subjects_list)

  File "/Users/sophie/repos/mne-study-template/05a-run_ica.py", line 128, in <genexpr>
    parallel(run_func(subject) for subject in config.subjects_list)

  File "/Users/sophie/repos/mne-study-template/05a-run_ica.py", line 93, in run_ica
    ica.fit(epochs_for_ica, picks=picks, decim=decim)

  File "</Users/sophie/anaconda/envs/mne/lib/python3.6/site-packages/mne/externals/decorator.py:decorator-gen-365>", line 2, in fit

  File "/Users/sophie/anaconda/envs/mne/lib/python3.6/site-packages/mne/utils/_logging.py", line 89, in wrapper
    return function(*args, **kwargs)

  File "/Users/sophie/anaconda/envs/mne/lib/python3.6/site-packages/mne/preprocessing/ica.py", line 460, in fit
    self._fit_epochs(inst, picks, decim, verbose)

  File "/Users/sophie/anaconda/envs/mne/lib/python3.6/site-packages/mne/preprocessing/ica.py", line 559, in _fit_epochs
    self._fit(data, self.max_pca_components, 'epochs')

  File "/Users/sophie/anaconda/envs/mne/lib/python3.6/site-packages/mne/preprocessing/ica.py", line 634, in _fit
    self.n_components)

TypeError: %i format: a number is required, not dict

File: 06a-apply_ica.py [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+                                           baseline=(None, 0), tmin=-0.5,
+                                           tmax=0.5)
+
+            ecg_average = ecg_epochs.average()
+
+            # XXX I had to lower the threshold for ctps (default 0.25), otherwise it does not
+            # find any components
+            # check how this behaves on other data
+            ecg_inds, scores = ica.find_bads_ecg(ecg_epochs, method='ctps',
+                                                 threshold=0.1)
```
Comment (SophieHerbst) [inline 06a-apply_ica.py]: @agramfort 
I had to reduce the threshold for ctps, otherwise to rather obviously outstanding components did not get selected.

**Comment by SophieHerbst** [comment]
@agramfort ok to merge?

**Comment by agramfort** [comment]
if it does the job for you sure. It's not working yet for SSPs though

**Comment by SophieHerbst** [comment]
what do you mean by its not working for SSPs? does it break? 
ICA and SSP where meant to function as parallel/ independent routes.

**Pull request merged**